File: Sanatorium.PatientService/DTO/PatientDTO.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Sanatorium.PatientService.DTO
{
	public class PatientDTO
	{
		public Guid Id { get; set; }
		public string FirstName { get; set; }
		public string LastName { get; set; }
		public string MiddleName { get; set; }
		public DateOnly BirthDate { get; set; }
		public string PhoneNumber { get; set; }

		public DateOnly DateRegistered { get; set; }
		public DateOnly? DateDischarged { get; set; }
		public bool Discharged { get; set; }
	}
}

File: Frontend/sanatorium/src/ApiDataClasses/PatientDTO.ts
export class PatientDTO{
    Id : string;
    FirstName: string;
    LastName : string;
    MiddleName : string;
    BirthDate : Date;
    PhoneNumber : string;

    DateRegistered : Date;
    DateDischarged : null | Date;
    Discharged : boolean;
    
    
    constructor(jsonData: any){
        this.Id = jsonData.Id;
        this.FirstName = jsonData.FirstName;
        this.LastName = jsonData.LastName,
        this.MiddleName = jsonData.MiddleName;
        this.BirthDate = new Date(jsonData.BirthDate);
        this.PhoneNumber = jsonData.PhoneNumber;
        this.DateRegistered = new Date(jsonData.DateRegistered);
        if(jsonData.DateDischarged && jsonData.DateDischarged !== ''){
            this.DateDischarged = new Date(jsonData.DateRegistered);
        }
        else{
            this.DateDischarged = null;
        }
        this.Discharged = jsonData.Discharged;
    }
}